src/popper/runner_host.py

```diff
@@ -229,6 +228,155 @@ def _find_container(self, cid):
         return None
 
 
+class PodmanRunner(StepRunner):
+
+    """Runs steps in podman on the local machine."""
+
+    def __init__(self, init_podman_client=True, **kw):
+        super(PodmanRunner, self).__init__(**kw)
+
+        self._spawned_containers = set()
+
+        if not init_podman_client:
+            return
+
+        try:
+            _, _, self._p_info = HostRunner._exec_cmd(["podman", "info"], logging=False)
+            self._p_version = HostRunner._exec_cmd(["podman", "version"], logging=False)
+        except Exception as e:
+            log.debug(f"Podman error: {e}")
+            log.fail("Unable to connect to podman, is it installed?")
+
+        log.debug(f"Podman info: {pu.prettystr(self._p_info)}")
+
+    def run(self, step):
+        """Executes the given step in podman."""
+        cid = pu.sanitized_name(step.id, self._config.wid)
+        container = self._find_container(cid)
+
+        if not container and self._config.reuse:
+            log.fail(
+                f"Cannot find an existing container for step '{step.id}' to be reused"
+            )
+
+        if container and not self._config.reuse and not self._config.dry_run:
+            cmd = ["podman", "rm", "-f", container]
+            HostRunner._exec_cmd(cmd, logging=False)
+            container = None
+
+        if not container and not self._config.reuse:
+            container = self._create_container(cid, step)
+
+        log.info(f"[{step.id}] podman start")
+
+        if self._config.dry_run:
+            return 0
+
+        self._spawned_containers.add(container)
+
+        cmd = ["podman", "start", "-a", container]
+        _, e, output = HostRunner._exec_cmd(cmd, logging=False)
```

Comment:
this is swallowing the step output. Instead, the code should allow logging.

Since this is not causing any failure on the unit tests, we can open an issue to add new tests for this and other runners, as a way of ensuring that when a step prints information to stdout, the loggers are indeed capturing this output. Would it be possible for you to open an issue for this?

Reply:
I added a print statement to show the step output. I did this because I needed a way to log the output when the start command gets executed and stream it to the terminal whether there is a failure or not. So it's sorta swallowing the output to release it later

I can open the issue, I'd need ideas on how to fix it though

Reply:
Sorry, I didn't understand what you mean. Can you please expand a bit on what you mean by "added print statement to show output"? Where do you have to explicitly print? That shouldn't be needed as output to stdout is generated by the `_exec_cmd` method via the logging framework. Removing `logging=False` or explicitly doing `logging=True` have that effect.

The new issue I mentioned (we'd open a new one) is related to adding tests that ensure that steps are printing to stdout when they are expected to (and test the opposite case as well). If we had that test already, it would have catched the above (where `logging=False` was given, and thus would have triggered a failure).